src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -68,11 +68,6 @@ public override bool Equals(object comparand)
                 return false;
             }
 
-            if (_address.GetHashCode() != address._address.GetHashCode())
-            {
-                return false;
```

Comment:
I imagine this was meant to be `if (GetHashCode() != address.GetHashCode()) return false` on the basis that the hash code is memoised and so this could be an effective short-circuit. Of course it could also be busy-work, but perhaps:

```C#
if (!hashNotComputed && !address.hashNotComputed && _hash != address._hash)
{
    return false;
}
```

Would retain the attempted use of the memoisation without either giving incorrect results, or doing more work when the hash hasn't been memoised.